src/Symfony/Component/Console/Helper/TableStyle.php

```diff
@@ -114,9 +123,13 @@ public function getVerticalBorderChar()
      * @param string $crossingChar
      *
      * @return $this
+     *
+     * @deprecated since version 4.1, to be removed in 5.0. Use {@link setCrossingChars()} instead.
      */
     public function setCrossingChar($crossingChar)
     {
+        @trigger_error(sprintf('%s() is deprecated since version 4.1 and will be removed in 5.0. Use setCrossingChars() instead.', __METHOD__), E_USER_DEPRECATED);
+
         $this->crossingChar = $crossingChar;
```

Comment:
should call `$this->setCrossingChars($crossingChar);` to update all properties

perhaps consider to not deprecate this method and make args in setCrossingChars required instead.

Reply:
1. Oups, bad rebase. It's fixed.
2. Personally I prefer with optional args. Does this https://github.com/symfony/symfony/pull/25456#discussion_r157340496 convince you?

Reply:
Mmh finally, I have change my opinion :) I think that `setCrossingChar()` can be conserved and used as a shortcut of `setCrossingChars()`. Moreover, that simplifies a lot this PR. Could you check my second commit? If you disagree, I can remote it.

Reply:
If you keeo the `setCrossingChar`, then maybe you should add `setTopLeft`, and so on. And then get rid of `setCrossingChars`.

Reply:
I don't think so. If you don't use a generic crossing char by setting setCrossingChar(), you have to set all kinds of crossing char. And it's better if we can do it with only 1 call of setCrossingChars() instead 9 different calls (1 cross + 4 corners + 4 mids).

Reply:
Isn't the idea of this PR to deprecate the "generic crossing char", in favor of using the specific ones ? The `setCorssingChar` should be imo just for the `0` (any "crossroad" delimiters). Using it as a generic should be deprecated.

Reply:
It was. But the https://github.com/symfony/symfony/pull/25456#discussion_r157344705 of @ro0NL looks relevant:
```php
// we can set specific chars:
setCrossingChars('┼', '┌', '┬', '┐', '┤', '┘', '┴', '└', '├');

// and we also could set generic char (useful for no-unicode):
setCorssingChar('+');

// without this last one, we should write:
setCrossingChars('+', '+', '+', '+', '+', '+', '+', '+', '+');
```